			   READLINE PATCH REPORT
			   =====================

Readline-Release: 6.1
Patch-ID: readline61-001

Bug-Reported-by:	guillaume.outters@free.fr
Bug-Reference-ID:	<20100105230441.70D171AA7F52@asterix.local>
Bug-Reference-URL:	http://lists.gnu.org/archive/html/bug-bash/2010-01/msg00017.html

Bug-Description:

Bash-4.1/Readline-6.1 introduced a hook function that allows applications
to rewrite or modify filenames read from the file system before comparing
them with a word to be completed.  The converted filename, if it matches,
needs to be inserted into the line buffer, replacing the original contents.

This fixes a completion bug on Mac OS X involving filenames containing
UTF-8 characters.

--- ./patchlevel.orig	2010-04-10 12:20:48.641085669 +0000
+++ ./patchlevel	2010-04-10 12:21:04.611086065 +0000
@@ -1,3 +1,3 @@
 # Do not edit -- exists only for use by patch
 
-0
+1
--- ./complete.c.orig	2010-04-10 12:20:48.651086558 +0000
+++ ./complete.c	2010-04-10 12:21:04.611086065 +0000
@@ -2138,7 +2138,7 @@
 	 All other entries except "." and ".." match. */
       if (filename_len == 0)
 	{
-	  if (_rl_match_hidden_files == 0 && HIDDEN_FILE (entry->d_name))
+	  if (_rl_match_hidden_files == 0 && HIDDEN_FILE (convfn))
 	    continue;
 
 	  if (convfn[0] != '.' ||
@@ -2219,7 +2219,7 @@
 		temp[dirlen++] = '/';
 	    }
 
-	  strcpy (temp + dirlen, entry->d_name);
+	  strcpy (temp + dirlen, convfn);
 	}
       else
 	temp = savestring (convfn);
